all_seaing_driver/all_seaing_driver/webcam_publisher.py

```diff
@@ -12,25 +12,33 @@ def __init__(self):
         super().__init__("webcam_publisher")
         self.publisher_ = self.create_publisher(Image, "webcam_image", 10)
         self.bridge = CvBridge()
+        self.timer = self.create_timer(1.0/30.0, self.publish_webcam_image)
+        self.cap = cv2.VideoCapture(0)
+
 
     def publish_webcam_image(self):
-        cap = cv2.VideoCapture(0)
-        while True:
-            ret, frame = cap.read()
-            if ret:
-                ros_image = self.bridge.cv2_to_imgmsg(frame, "bgr8")
-                self.publisher_.publish(ros_image)
-            if cv2.waitKey(1) & 0xFF == ord("q"):
-                break
-        cap.release()
-        cv2.destroyAllWindows()
+        ret, frame = self.cap.read()
+        if ret:
+            ros_image = self.bridge.cv2_to_imgmsg(frame, "bgr8")
+            self.publisher_.publish(ros_image)
+        if cv2.waitKey(1) & 0xFF == ord("q"):
```

Comment:
Let's see if the node creates any windows? If not, we can probably remove the waitkey stuff and also the cv2.destroyAllWindows